src/main/scala/zio/kafka/admin/AdminClient.scala

```diff
@@ -593,7 +597,7 @@ object AdminClient {
         description.members.asScala.map(MemberDescription.apply).toList,
         description.partitionAssignor,
         ConsumerGroupState(description.state),
-        Node(description.coordinator()),
+        Node(description.coordinator()).get,
```

Comment:
Don't know if this is the correct way here.
Can `ConsumerGroupDescription` sometime have no coordinator node. I would guess no

Reply:
I wouldnt guess that tbh. Distributed system here. Can you look this model in Conduktor desktop, see if nullable or not?
Maybe a coordinator could die without a reelection to have occured in the meantime.
I would always make this nullable. We must be resilient and manage such cases.

Reply:
```
class ConsumerGroupDescription(
  groupId: String,
  simple: Boolean,
  members: List<MemberDescription> = mutableListOf(),
  assignor: String? = null,
  state: ConsumerGroupState? = null,
  coordinator: Node? = null
)
```
So you right, let's go for option

Reply:
What's interesting too is that in conduktor desktop you handle the jNone.noNode this way:
```
sealed interface KNode {
  object InError : KNode
  class Healthy(val id: String, val port: Int, val rack: String?) : KNode

  companion object {
    fun fromKafka(node: Node?) = if (node == null || node.isEmpty) InError else Healthy(
      node.idString(),
      node.port(),
      node.rack()
    )
  }
}
```
I think we should have the same approach in ZIO because if when the lib users will encounter a `Seq[Option[Node]]`, he will  probably flatten it to get a `Seq[Node]` and then we loose some information about having a node but in error.

Reply:
I would not model an `InError` at this point in devtools and zio-kafka, I would go for using `Option[Node]`for the fields where it makes sense to have no value: leader and coordinator.
For the fields representing a list of nodes (like replicas), it should never contain `InError` values. Only valid node identifier, whether they are available or not. And thus be represented as `List[Node]`

Reply:
Ok I updated the PR to do that

Reply:
Ok I run some local test after running a cluster and killing a node I got:
```
app-embedded[ERROR] Execution Error: Effect failure io.conduktor.devtools.api.gql.ErrorInfo$InternalError: ErrorCore: INTERNAL_ERROR. Message: Unexpected Kafka error: NoNode node not expected among topic replicas. jPartition: (partition=0, leader=127.0.0.1:9092 (id: 1 rack: null), replicas=127.0.0.1:9092 (id: 1 rack: null), :-1 (id: 3 rack: null), 127.0.0.1:9093 (id: 2 rack: null), isr=127.0.0.1:9092 (id: 1 rack: null), 127.0.0.1:9093 (id: 2 rack: null))
```
So I got back to my question.
Should Node.apply(jNode) should return Option[Node] or just Node where node would be a sealed trait with InvalidNode/Node.
If we go to Option[Node] should we then flatten or return a List[Option[Node]]. If we flatten we loose information about the number of replicas before a/some node failure. And this is an issue because the way we know a partition is undereplicated is by comparing isr.length with replicas.length.
In this case, if we flatten we can't detect the partition is under replicated. I suggest we go back on commit 49a579c

Reply:
In this case it seems node is not null but hostname/port is missing.

I would stick as much as possible to the kafka model and make hostName and port optional in Node, so we can convey the information that there is a replica on brokerId x, but this broker is currently offline.

Does that make sense?

Reply:
Ok you can also `NoNode` into `isr`:
```
app-embedded[ERROR] Execution Error: Effect failure io.conduktor.devtools.api.gql.ErrorInfo$InternalError: ErrorCore: INTERNAL_ERROR. Message: Unexpected Kafka error: NoNode node not expected among topic in sync replicas. jPartition: (partition=1, leader=null, replicas=:-1 (id: 3 rack: null), isr=:-1 (id: 3 rack: null))
```

Reply:
good to know (I guess kafka knows it is inSync but unavailable and will remove it from isr when you publish enough messages)